File: LeetCode/203. Remove Linked List Elements.cpp
/***  KNOWLEDGE IS POWER  ***/

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeElements(ListNode* head, int T) {
        if(!head) 
            return nullptr;
        head -> next = removeElements(head -> next, T) ;
        return head -> val == T ? head -> next : head ;
    }
};

File: LeetCode/203. Remove Linked List Elements.py
### KNOWLEDGE IS POWER  ###

# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def removeElements(self, head, T):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        if not head: 
            return None
        head.next = self.removeElements(head.next, T);
        return head.next if head.val == T else head
        
